packages/blocks/src/page-block/edgeless/selection-manager/shape.ts

```diff
@@ -0,0 +1,83 @@
+import type { SelectionEvent, ShapeMouseMode } from '../../../__internal__';
+import { assertExists } from '../../../__internal__';
+import { SelectionController } from './index';
+import type { SelectionArea } from '../selection-manager';
+
+export class ShapeSelectionController extends SelectionController<ShapeMouseMode> {
+  readonly mouseMode = <ShapeMouseMode>{
+    type: 'shape',
+    color: 'black',
+    shape: 'rectangle',
+  };
+
+  private _draggingShapeBlockId: string | null = null;
```

Comment:
What if we have multiple shapes being dragged together?

Reply:
oh, this is only for creating a new shape(drag from a point to another pont)